src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreator.java

```diff
@@ -145,8 +145,22 @@ public static void createSystemHookWhenMissing(GitLabServer server,
         }
     }
 
+    /**
+     * @deprecated use {@link #getHookUrl(String,boolean)} instead
+     */
+    @Deprecated
     public static String getHookUrl(boolean isWebHook) {
-        String rootUrl = Jenkins.get().getRootUrl();
+        return getHookUrl(null, isWebHook);
+    }
+
+    /**
+     * @param hooksRootUrl the root URL to use in the hook URL.
+     *        If {@code null} or empty, {@link Jenkins#getRootUrl()} will be used instead.
+     * @param isWebHook {@code true} to get the webhook URL, {@code false} for the systemhook URL
+     * @return a webhook or systemhook URL
+     */
+    public static String getHookUrl(String hooksRootUrl, boolean isWebHook) {
+        String rootUrl = (hooksRootUrl == null) ? Jenkins.get().getRootUrl() : hooksRootUrl;
```

Comment:
I might be blind, do we have tests for both cases here?

Reply:
> do we have tests for both cases here?

Yes, in `GitLabHookCreatorParameterizedTest`, `hookUrl()` vs. `hookUrlFromCustomRootUrl()`. They both use the same set of test parameters, but:
- the former uses the URL param as a Jenkins root URL, with `hooksRootUrl=null` (that's a test which was already there and which I've only slightly modified)
- the later uses it as a `hooksRootUrl`, with a dummy value for the Jenkins root URL (that's the test I've added)